collection_test.go

```diff
@@ -728,3 +691,74 @@ func ExampleCollectionSpec_LoadAndAssign() {
 	// Output: SocketFilter
 	// Array
 }
+
+var collectionSpecCmpOptions = cmp.Options{
+	cmp.Comparer(func(a, b asm.Metadata) bool {
+		return a == b
+	}),
+	cmp.Comparer(func(a, b *Program) bool {
+		return a == b
+	}),
+	cmp.Comparer(func(a, b *btf.Spec) bool {
+		// No way to compare a btf.Spec at the moment, this is enough for now.
+		return a == b
+	}),
+	// Empty MapSpec.Extra are equal.
+	cmp.Comparer(func(a, b bytes.Reader) bool {
+		if a.Len() == 0 && b.Len() == 0 {
+			return true
+		}
+		return false
+	}),
+}
+
+var collectionSpecIsEqual = qt.CmpEquals(collectionSpecCmpOptions...)
+
+func mustNewCollection(t *testing.T, spec *CollectionSpec, opts *CollectionOptions) *Collection {
+	t.Helper()
+
+	coll, err := newCollection(t, spec, opts)
+	if err != nil {
+		t.Fatal("Load collection:", err)
+	}
+
+	return coll
+}
+
+func mustNewCollectionSkipUnsupported(t *testing.T, spec *CollectionSpec, opts *CollectionOptions) *Collection {
+	t.Helper()
+
+	if spec.ByteOrder != internal.NativeEndian {
+		t.Skip("Spec has foreign byte order", spec.ByteOrder)
+	}
+
+	coll, err := newCollection(t, spec, opts)
+	testutils.SkipIfNotSupported(t, err)
+	if err != nil {
+		t.Fatal("Load collection:", err)
+	}
+
+	return coll
+}
+
+func newCollection(t *testing.T, spec *CollectionSpec, opts *CollectionOptions) (*Collection, error) {
+	t.Helper()
+
+	if opts == nil {
+		opts = new(CollectionOptions)
+	}
+
+	cpy := spec.Copy()
+	coll, err := NewCollectionWithOptions(spec, *opts)
+	if coll != nil {
+		t.Cleanup(func() { coll.Close() })
+	}
+
+	qt.Assert(t, spec, collectionSpecIsEqual, cpy)
```

Comment:
Maybe add a comment explaining why this check exists. It's an innocuous but powerful one-liner. :)